packages/devtools/lib/src/ui/plotly.dart

```diff
@@ -0,0 +1,237 @@
+@JS()
+library plotly;
+
+import 'package:js/js.dart';
+
+@JS()
+external void hookupPlotlyClick(String domName, Function jsFunction);
+
+@JS()
+external void myExtendTraces(String domName, List x0, List x1, List x2, List x3, List y0, List y1, List y2, List y3, List traces);
+
+@JS()
+class Plotly {
+  external static void newPlot(String domName, List<Data> traces, [Layout layout, Configuration config]);
+  external static void extendTraces(String domName, TraceData traceData, List<int> traceindex);
+  external static void update(String domName, Data data, Layout layout, [List<int> traceindex]);
+}
+
+@JS()
+@anonymous
+class TraceData {
+  external List<List> get x;
+  external List<List> get y;
+
+  external factory TraceData({List<List> x, List<List> y});
```

Comment:
nit: move constructor to the top of the class

Reply:
Done everywhere in this file.